File: examples/java7-example/src/main/java/nz/bradcampbell/paperparcel/java7example/State.java
package nz.bradcampbell.paperparcel.java7example;

import java.util.Date;
import nz.bradcampbell.paperparcel.PaperParcel;
import nz.bradcampbell.paperparcel.PaperParcelable;
import nz.bradcampbell.paperparcel.TypeAdapters;

@PaperParcel
@TypeAdapters(DateTypeAdapter.class)
public final class State extends PaperParcelable {
  public static final PaperParcelable.Creator<State> CREATOR =
      new PaperParcelable.Creator<>(State.class);

  private final int count;

  public Date modificationDate;

  public State(int count) {
    this.count = count;
  }

  public int getCount() {
    return count;
  }
}

File: examples/kotlin-example/src/main/kotlin/nz/bradcampbell/paperparcel/kotlinexample/State.kt
package nz.bradcampbell.paperparcel.kotlinexample

import nz.bradcampbell.paperparcel.PaperParcel
import nz.bradcampbell.paperparcel.PaperParcelable
import nz.bradcampbell.paperparcel.TypeAdapters
import java.util.Date

@PaperParcel
@TypeAdapters(DateTypeAdapter::class)
data class State(val count: Int, val modificationDate: Date) : PaperParcelable {
  companion object {
    @JvmField val CREATOR = PaperParcelable.Creator(State::class.java)
  }
}
